sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala

```diff
@@ -232,6 +232,16 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
     visitMultipartIdentifier(ctx.collationName)
   }
 
+  protected def visitCollationSpecList(
+      ctx: java.util.List[CollationSpecContext]): Option[String] = {
+    ctx.asScala.headOption.map(visitCollationSpec)
+  }
+
+  override def visitCollationSpec(ctx: CollationSpecContext): String = withOrigin(ctx) {
```

Comment:
How are these two methods related to `DataTypeAstBuilder`? They are for Spark DDL commands and should be put in `AstBuilder` I think, the same as other CREATE TABLE/VIEW clauses.

Reply:
Makes sense, moved these methods to AstBuilder.